File: solutions/julia/difference-of-squares/1/difference-of-squares.jl
"Square the sum of the first `n` positive integers"
square_of_sum(n) = (n * (n + 1) * (2 * n + 1)) / 6


"Sum the squares of the first `n` positive integers"
function sum_of_squares(n)
    ((n * (n + 1)) >> 1)^2

end

"Subtract the sum of squares from square of the sum of the first `n` positive ints"
difference(n) = square_of_sum(n) - sum_of_squares(n)

File: solutions/typescript/difference-of-squares/1/difference-of-squares.ts
export class Squares {
    #n: number;
    constructor(count: number) {
        this.#n = count;
    }

    get sumOfSquares(): number {
        return (this.sum * (2 * this.#n + 1)) / 3;
    }

    get squareOfSum(): number {
        return this.sum ** 2;
    }

    get difference(): number {
        return Math.abs(this.squareOfSum - this.sumOfSquares);
    }

    get sum(): number {
        return (this.#n * (this.#n + 1)) / 2;
    }
}
